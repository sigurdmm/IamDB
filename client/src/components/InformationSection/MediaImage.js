import PropTypes from 'prop-types';
import React from 'react';
import defaultImage from './no-image-found.jpg';

const MediaImage = ({ image = null }) => <div className='mediaimage'>
  <img className='image__item' src={image || defaultImage }/>
</div>;

MediaImage.propTypes = {
  image: PropTypes.string,
};

export default MediaImage;
