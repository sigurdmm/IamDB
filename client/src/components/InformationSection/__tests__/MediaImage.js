import { render } from 'enzyme/build';
import MediaImage from '../MediaImage';

const mockedImage = {
  thumbnails: {
    small: 'https://example.com/test.jpg',
  },
  noThumbnails: {},
};

describe('<MediaImage/>', () => {
  it('should render img, when link exists', () => {
    const wrapper = render(<MediaImage image={mockedImage.thumbnails.small}/>);

    expect(wrapper).toMatchSnapshot();
  });
  it('should render default img, when link does not exist ', () => {
    const wrapper = render(<MediaImage image={mockedImage.noThumbnails.small}/>);

    expect(wrapper).toMatchSnapshot();
  });
});
