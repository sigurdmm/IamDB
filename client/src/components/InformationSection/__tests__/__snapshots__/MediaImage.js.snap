// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MediaImage/> should render default img, when link does not exist  1`] = `
<div
  class="mediaimage"
>
  <img
    class="image__item"
    src="no-image-found.jpg"
  />
</div>
`;

exports[`<MediaImage/> should render img, when link exists 1`] = `
<div
  class="mediaimage"
>
  <img
    class="image__item"
    src="https://example.com/test.jpg"
  />
</div>
`;
